pkg/apis/wildfly/v1alpha1/wildflyserver_types.go

```diff
@@ -130,6 +134,26 @@ type WildFlyServerList struct {
 	Items           []WildFlyServer `json:"items"`
 }
 
+// ConfigMapSpec represents a ConfigMap definition with a name and a mount path
+// +k8s:openapi-gen=true
+type ConfigMapSpec struct {
+	Name      string `json:"name"`
+	MountPath string `json:"mountPath,omitempty"`
+}
+
+// ResolveMountPath resolves the final mount path for the current ConfigMapSpec
+func (in *ConfigMapSpec) ResolveMountPath() string {
+	result := path.Join(constants.ConfigMapsDir, in.Name)
+	if in.MountPath != "" {
+		if path.IsAbs(in.MountPath) {
+			result = in.MountPath
+		} else {
+			result = path.Join(constants.JBossHome, in.MountPath)
+		}
+	}
+	return result
```

Comment:
at this place, we could have a sanity check that there is no `:` in the resulting path

Reply:
Now we are using openapi validation at CR level